src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/custom.py

```diff
@@ -108,20 +78,64 @@ def cli_redis_create(cmd, client,
     :param enable_non_ssl_port: If the value is true, then the non-ssl redis server port (6379) will be enabled.
     :param tenant_settings: Json dictionary with tenant settings
     :param shard_count: The number of shards to be created on a Premium Cluster Cache.
+    :param minimum_tls_version: Optional: requires clients to use a specified
+     TLS version (or higher) to connect (e,g, '1.0', '1.1', '1.2'). Possible
+     values include: '1.0', '1.1', '1.2'
     :param subnet_id: The full resource ID of a subnet in a virtual network to deploy the redis cache in. Example format /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1
     :param static_ip: Required when deploying a redis cache inside an existing Azure Virtual Network.
+    :param zones: A list of availability zones denoting where the resource needs to come from.
+    :param tags: Json dictionary with Resource tags. Example : {\"testKey\":\"testValue\"}
     """
     from azure.mgmt.redis.models import RedisCreateParameters, Sku
+    # pylint: disable=too-many-function-args
     params = RedisCreateParameters(
-        location=location,
-        sku=Sku(sku, vm_size[0], vm_size[1:]),
-        tags=tags,
+        Sku(sku, vm_size[0], vm_size[1:]),
+        location,
         redis_configuration=redis_configuration,
         enable_non_ssl_port=enable_non_ssl_port,
         tenant_settings=tenant_settings,
         shard_count=shard_count,
+        minimum_tls_version=minimum_tls_version,
         subnet_id=subnet_id,
-        static_ip=static_ip)
+        static_ip=static_ip,
+        zones=zones,
+        tags=tags)
     return client.create(resource_group_name, name, params)
 
+
+# pylint: disable=unused-argument
+def cli_redis_create_server_link(cmd, client, primary_cache, secondary_cache):
+    redis_client = cf_redis(cmd.cli_ctx)
+    primary_cache_resource = get_cache_from_resource_id(redis_client, primary_cache)
+    primary_cache_resource_group = get_resource_group_name_by_resource_id(primary_cache)
+    secondary_cache_resource = get_cache_from_resource_id(redis_client, secondary_cache)
+
+    from azure.mgmt.redis.models import RedisLinkedServerCreateParameters
+    params = RedisLinkedServerCreateParameters(secondary_cache_resource.id,
+                                               secondary_cache_resource.location, 'Secondary')
+    return client.create(primary_cache_resource_group, primary_cache_resource.name,
+                         secondary_cache_resource.name, params)
+
+
+def cli_redis_list_cache(client, resource_group_name=None):
+    cache_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+    return list(cache_list)
+
+
+def get_resource_group_name_by_resource_id(resource_id):
```

Comment:
use `mrestazure.tools.parse_resource_id` instead. The code base has examples of its usage.